test/unit/solver/test_cuda_linear_solver.cpp

```diff
@@ -103,22 +90,12 @@ void verify_gpu_against_cpu()
   std::vector<double> cpu_x = linearSolverGenerator<
       Group10000VectorMatrix,
       Group10000SparseVectorMatrix,
-      micm::LinearSolver<double, Group10000SparseVectorMatrix>>(
-      [](const Group10000SparseVectorMatrix<double>& matrix,
-         double initial_value) -> micm::LinearSolver<double, Group10000SparseVectorMatrix> {
-        return micm::LinearSolver<double, Group10000SparseVectorMatrix>{ matrix, initial_value };
-      },
-      10000);
+      micm::LinearSolver<Group10000SparseVectorMatrix>>(10000);
```

Comment:
why don't we need the lambda function here any more?

Reply:
It's not clear to me that we ever needed it here. we can call the construction directly in the linear solver function. I can put these back if we like having the lamdas with explicit creation

Reply:
Thanks. @mattldawson do you recall why we use the lambda function argument before?

Reply:
It was for the JITed version of things. We needed to pass the jit compiler around, but, I made the jit compiler into a singleton and now the interface to all solvers and their components are the same